python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -896,6 +896,114 @@ def corr(self, method="pearson", min_periods=1):
 
         return res
 
+    def cov(self, min_periods=None, ddof=1):
+        """
+        Compute pairwise covariance of columns, excluding NA/null values.
+
+        Parameters
+        ----------
+        min_periods: int, optional
+            Minimum number of observations required per pair of columns
+            to have a valid result.
+
+        ddof: int, optional
+            Delta degrees of freedom, default is 1.
+
+        Returns
+        ----------
+        DataFrame
+            Covariance matrix.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> gdf = cudf.DataFrame({
+        ...             "id": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+        ...             "val1": [5, 4, 6, 4, 8, 7, 4, 5, 2],
+        ...             "val2": [4, 5, 6, 1, 2, 9, 8, 5, 1],
+        ...             "val3": [4, 5, 6, 1, 2, 9, 8, 5, 1]})
+        >>> gdf
+        id  val1  val2  val3
```

Comment:
I think this will need to align correctly for doctests (#9815) to pass. You might need to double-check how the printout aligns when you run this snippet in a terminal for the doctest below as well.
```suggestion
          id  val1  val2  val3
```